src/Sparse/Matrix.jl

```diff
@@ -952,6 +952,162 @@ function Base.insert!(A::SMat{T}, i::Integer, B::SRow{T}) where T
+
+Deletes zero rows of $A$ in place.
+"""
+function delete_zero_rows!(A::SMat{T}, s=1) where T #where s denotes the first column where we wanna start
+  for i=A.r:-1:s
+      if A[i].pos == []
+          deleteat!(A.rows, i); A.r-=1
+      end
+  end
+  return A
+end
+
+@doc Markdown.doc"""
+  empty_col!(A::SMat{T}, TA::SMat{T}, j::Int) -> SMat{T}
+
+Deletes non-zero entries in $j$-th column of $A$ in place.
+"""
+function empty_col!(A::SMat{T}, TA::SMat{T}, j::Int) where T #only deletes entries in column j, output same size as input
+  for row in TA[j].pos 
+      i = findfirst(isequal(j), A[row].pos)
+      deleteat!(A[row].pos, i) ; deleteat!(A[row].values, i)
+  end
+  A.nnz -=length(TA[j].pos)
+  return A
+end
+
+@doc Markdown.doc"""
+  empty_cols!(A::SMat{T}, TA::SMat{T}, J) -> SMat{T}
+
+Deletes non-zero entries in columns with indices in $J$ of $A$ in place.
+"""
+function empty_cols!(A::SMat{T}, TA::SMat{T}, J) where T
+  for j in J
+      empty_col!(A, TA, j)
+  end
+  return A
+end
+
+################################################################################
+#
+#  Row/Col operations in matrix
+#
+################################################################################
+@doc Markdown.doc"""
+  function scale_col!(A::SMat{T}, j::Int, c::T) -> SMat{T}
+
+Returns $A$ after scaling j-th column in place using transposed matrix.
+"""
+function scale_col!(A, TA, j, c) #A[_j]->c*A[_,j]
+  for i in TA[j].pos
+      idx_j = findfirst(isequal(j), A[i].pos)
+      A[i].values[idx_j]*=c
+  end
+  return A
+end
+
+@doc Markdown.doc"""
+    add_scaled_row!(A::SMat{T}, i::Int, j::Int, c::T) -> SMat{T}
+
+Returns $A$ after add_scaled_row!(Ai::SRow{T}, Aj::SRow{T}, c::T) in $A$.
+"""
+function add_scaled_row!(A::SMat{T}, i::Int, j::Int, c::T) where T
+  A.nnz = A.nnz - length(A[j])
+  add_scaled_row!(A[i], A[j], c)
+  A.nnz = A.nnz + length(A[j])
+  return A
+end
+
+@doc Markdown.doc"""
+    add_scaled_col!(A::SMat{T}, i::Int, j::Int, c::T) -> SMat{T}
+
+As add_scaled_row!(A::SMat{T}, i::Int, j::Int, c::T) but with columns of $A$.
+"""
+function add_scaled_col!(A::SMat{T}, i::Int, j::Int, c::T) where T 
+  @assert c != 0
+  @assert 1 <= i <= ncols(A) && 1 <= j <= ncols(A)  
+  for r in A.rows
+    if i in r.pos
+      i_i = findfirst(isequal(i), r.pos) #changed
+      val_i = r.values[i_i]
+      if j in r.pos
+        i_j = findfirst(isequal(j), r.pos) #changed
+        val_j = r.values[i_j]
+
+        r.values[i_j] += c*r.values[i_i]
+      else
+        k = searchsortedfirst(r.pos, j)
+        insert!(r.pos, k, j)
+        insert!(r.values, k, c*r.values[i_i])
+        A.nnz+=1 #necessary in matrices
+      end
+    end
+  end
+  return A
+end
+
+@doc Markdown.doc"""
+    add_scaled_col!(A::SMat{T}, TA::SMat{T}, i::Int, j::Int, c::T) -> SMat{T}
+
+As add_scaled_row!(A::SMat{T}, i::Int, j::Int, c::T) but with columns of $A$.
+"""
+function add_scaled_col!(A::SMat{T}, TA::SMat{T}, i::Int, j::Int, c::T) where T  #A[_j]->c*A[_,i]+A[_j]
```

Comment:
Afterwards `TA` will in general not be the transpose of `A` anymore. Is this intended?

Reply:
Yes. In the preprocessing function where I use this TA is changed afterwards.
Unfortunately preprocessing for fmpz isn't working at the moment, I'll change the aux function, after finding the error.